File: scala-3.6/template-renderer/src/test/scala/com/rbleggi/templaterenderer/renderers/PDFRendererTest.scala
package com.rbleggi.templaterenderer.renderers

import org.scalatest.funsuite.AnyFunSuite

class PDFRendererTest extends AnyFunSuite {

  test("should generate valid PDF content") {
    val pdfBytes = new PDFRenderer().render(Map("title" -> "PDF Test", "content" -> "Testing PDF generation"))
    assert(pdfBytes.nonEmpty, "Generated PDF content should not be empty")
  }
}

File: kotlin/template-renderer/src/test/kotlin/com/rbleggi/templaterenderer/service/renderer/PdfRendererTest.kt
package com.rbleggi.templaterenderer.service.renderer

import com.rbleggi.templaterenderer.model.Document
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class PdfRendererTest {

    private val document = Document("Title", "Content")

    @Test
    fun `render produces a non-empty pdf`() {
        assertTrue(PdfRenderer().render(document).isNotEmpty())
    }

    @Test
    fun `render produces bytes with the pdf header`() {
        assertTrue(String(PdfRenderer().render(document)).startsWith("%PDF-"))
    }

    @Test
    fun `different documents produce different pdfs`() {
        val other = Document("Other", "Different")
        assertFalse(PdfRenderer().render(document).contentEquals(PdfRenderer().render(other)))
    }
}
